### PR TITLE
User/kyrader/add publish job

### PR DESCRIPTION
**Pull request by kyle-rader**
# Setup Microsoft ESRP Signing for `.deb` installer
This adds a new workflow `build-installers.yml`, where hopefully we can either migrate all installer builds in the future (or rename and split them across the platforms.)

This workflow is heavily dependent on a lot of internal Microsoft infrastructure. What we have so far results in a GPG signed `.deb` for GCM-Core. This is currently uploaded as an artifact on the workflow itself but it not yet published. The 3rd and last build job yet to be added is using setting up yet another internal tool with the right security to use our internal publishing service.

Phases:
- [x] (linux agent) Build the Linux installer
- [x] (windows agent) Sign the Linux Installer
- [ ] (linux agent) Publish the Linux Installer to https://packages.microsoft.com - This phase I will do in a following pull request.

### CODE REVIEW

**Comment by kyle-rader** [comment]
@mjcheetham I'll still need to push a commit disabling this workflow from running on master PR's. It might still be nice to have the unsigned .deb built and maybe I can figure out how to extract that step as a re-usable action within this repo.

**Comment by kyle-rader** [comment]
@derrickstolee @mjcheetham  This is read for re-review, let's merge this in and I'll iterate on the second workflow to publish signed deb's from a release.

File: .github/workflows/build-signed-deb.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+name: "Build Signed Debian Installer"
+
+on:
+  push:
+    branches: [ release ]
```
Comment (derrickstolee) [inline .github/workflows/build-signed-deb.yml:5]: Do we want this on push, or...

```yaml
on:
  release: [released]
```

**Review by derrickstolee — COMMENTED** [review]
Some of this will just need to be run for real to see what happens.

File: .github/workflows/build-installers.yml [hunk 2]
```diff
@@ -1,36 +0,0 @@
-name: Build-Installers
-
-on:
-  push:
-    branches: [ master, release ]
```
[the remark this answers is not in the document]
Comment (derrickstolee) [reply .github/workflows/build-installers.yml:5]: Actually, I don't think we should change it at all.

**Comment by derrickstolee** [comment]
Replaced by #201.

File: .github/workflows/build-installers.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kyle-rader) [reply .github/workflows/build-installers.yml:5]: Ya keeping this just for building the unsigned installer for PRs is good.

**Pull request closed without merging**